the-thing-is.js

```diff
@@ -37,69 +39,78 @@ the.last = null
 var comparisons = {
-  is: function(whatYouExpect) {
-
-    var its = true
-    var expectation = null
-
-    // the(thing).is()
-    // whatYouExpect is undefined or null -- so check mere presence of a thing
-    if ( is.not.present(whatYouExpect) )
-      its = is.present(the.last.thing)
-
-    // 'present' -- single boolean check
-    // the(thing).is('integer') // true/false
-    // the(thing).is('borkborkbork') // throw
-    else
-      if ( is.string(whatYouExpect) )
-        its = booleanCheck(whatYouExpect)
-
-
-    // ['present', 'number'] -- array of boolean
-    // the(thing).is(['present', 'integer'])
-    else
-      if ( is.array(whatYouExpect) )
-        for (var i = 0; i < whatYouExpect.length && its == true; i++) {
-          expectation = whatYouExpect[i]
-          if (is.string(expectation))
-            its = booleanCheck(expectation)
-          else
-            its = subjectCheck(expectation)
-
-          if (!its)
-            the.last.error = '' + the.last.thing + ' is not ' + whatYouExpect[i];
-        }
-
-    // {greaterThan:0} -- configuration object with single condition
-    // {present:true, integer:true, greaterThan:0} -- unreliable due to unreliable hash key order
-    // [{present:true}, {integer:true}, {greaterThan:0}] -- reliable, but verbose
-    // ['present', 'integer', {greaterThanOrEqualTo:0}] -- preferred
-
-    return its
-  },
+  is: checkExpectations,
   isnt: function(whatYouExpect) {
     return !this.is(whatYouExpect)
   }
 }
 
+// recursive method of checking the expectations defined by the user
+function checkExpectations(whatYouExpect, subject) {
+
+  var its = true
+
+  subject = subject || the.last.thing
+
+  // the(thing).is()
+  // whatYouExpect is undefined or null -- so check mere presence of a thing
+  if ( is.not.present(whatYouExpect) )
+    its = is.present(the.last.thing)
+
+  // 'present' -- single boolean check
+  // the(thing).is('integer') // true/false
+  // the(thing).is('borkborkbork') // throw
+  else
+    if ( is.string(whatYouExpect) )
+      its = booleanCheck(whatYouExpect, subject)
+
+
+  // ['present', 'number'] -- array of boolean
+  // the(thing).is(['present', 'integer'])
+  else
+    if ( is.array(whatYouExpect) )
+      whatYouExpect.every(function(expectation){
+        if (is.string(expectation))
+          its = booleanCheck(expectation, subject)
+        else
+          its = subjectCheck(expectation, subject)
+        if (!its)
+          the.last.error = '' + subject + ' is not ' + expectation;
+        return its
+      })
+
+  // potentially deeply nested objects
+  else
+    if ( is.plainObject(whatYouExpect) ) {
+      its = Object.keys(whatYouExpect).every(function(key){
+        return checkExpectations(whatYouExpect[key], subject[key]);
+      })
+    }
+
+  return its
+}
+
 // simple yes/no type comparisons against an expectation
-function booleanCheck(expectation) {
+function booleanCheck(expectation, subject) {
+  subject = subject || the.last.thing
```

Comment:
Is there a better pattern that this? Thinking that an object shared between these methods would be useful. Or perhaps an object that holds these methods and the subject. Would need to respect function scope though.